plaidml/edsl/edsl_test.cc

```diff
@@ -46,6 +46,44 @@ Tensor Softmax(const Tensor& X) {
   return E / N;
 }
 
+TEST(CppEdsl, LowerPrecision) {
+  auto A = Placeholder(DType::FLOAT64, {3, 3});
+  auto C = A + 1 + 2.0;
+
+  Program program("lower_precision", {C}, DType::FLOAT32, DType::INT32);
+
+  // consatnt_types pass doesn't lower precision to float32/int32 until program is compiled
```

Comment:
typo: "constant_types"

But I think we should make the constant types pass happen during the construction of a program, not afterwards.

Reply:
done